src/scratch_comment_icon.js

```diff
@@ -0,0 +1,221 @@
+import * as Blockly from 'blockly/core';
+import {ScratchCommentBubble} from './scratch_comment_bubble.js';
+
+class ScratchCommentIcon extends Blockly.icons.Icon {
+  constructor(sourceBlock) {
+    super(sourceBlock);
+    this.sourceBlock = sourceBlock;
+    this.commentBubble = new ScratchCommentBubble(this.sourceBlock, this.getAnchorPoint());
+    Blockly.Events.setGroup(true);
+    Blockly.Events.fire(
+      new (Blockly.Events.get('block_comment_create'))(
+        this.commentBubble
+      ),
+    );
+    this.onTextChangedListener = this.onTextChanged.bind(this);
+    this.onSizeChangedListener = this.onSizeChanged.bind(this);
+    this.onCollapseListener = this.onCollapsed.bind(this);
+    this.commentBubble.addTextChangeListener(this.onTextChangedListener);
+    this.commentBubble.addSizeChangeListener(this.onSizeChangedListener);
+    this.commentBubble.addOnCollapseListener(this.onCollapseListener);
+    this.repositionAfterRender = true;
+
+    // We need to force a render of our parent block and wait for it to complete, because at this
+    // point, we don't know our own position (needed to calculate the anchor point and position
```

Comment:
I'm a bit confused about what's going on here - might be better to talk synchronously about this. Who is "we" in "we don't know our own position"?

Reply:
"We" is the icon here, but yeah happy to chat offline. This behavior was surprisingly and possibly technically a bug, but it's as described in the comment - the icon gets instantiated before the block it's attached to is fully laid out/rendered/initialized.

Reply:
Cleaned this up!